File: models/bucket.go
package models

import "time"

// Bucket representa un contenedor lógico de archivos, análogo a un bucket de
// S3/Firebase Storage. Cada bucket pertenece a un "owner" (el user_id
// asociado a la API key que lo creó).
type Bucket struct {
	ID        string    `json:"id" db:"id"`
	Name      string    `json:"name" db:"name"`
	Owner     string    `json:"owner" db:"owner"`
	IsPublic  bool      `json:"is_public" db:"is_public"`
	CreatedAt time.Time `json:"created_at" db:"created_at"`
}

// CreateBucketRequest es el payload de POST /api/buckets.
type CreateBucketRequest struct {
	Name     string `json:"name"`
	IsPublic bool   `json:"is_public"`
}

File: src/GabichoStorage.Api/Entities/Bucket.cs
namespace GabichoStorage.Api.Entities;

/// <summary>
/// Contenedor lógico de archivos, análogo a un bucket de S3/Firebase Storage.
/// Pertenece a un "owner" (el UserId de la API key que lo creó).
/// </summary>
public class Bucket
{
    public Guid Id { get; set; }

    public required string Name { get; set; }

    public required string Owner { get; set; }

    public bool IsPublic { get; set; }

    public DateTimeOffset CreatedAt { get; set; }

    public List<FileObject> Files { get; set; } = [];
}
